server/src/main/java/server/completion/CompletionProvider.java

```diff
@@ -1,158 +1,111 @@
+            index--;
+        }
+        return path.reverse().toString();
+    }
+
+    public static CompletionProvider getInstance() {
+        if(instance == null) {
+            instance = new CompletionProvider();
+        }
+        return instance;
+    }
 
     /**
      * This method takes the completionParams and determines enough context to give a list of
      * potential completionItems. These items will appear as autocomplete options for the user.
      *
-     * @param workspaceRoot    The path from the workspace root to the current file.
      * @param completionParams A object containing a TextDocumentIdentifier,
      *                         Position, and CompletionContext
      * @return A list of CompletionItems that can be used for autocomplete options for the user.
      */
-    public static CompletableFuture<Either<List<CompletionItem>, CompletionList>> getCompletion(
-            Path workspaceRoot, CompletionParams completionParams) {
+    public CompletableFuture<Either<List<CompletionItem>, CompletionList>> getCompletion(
+            CompletionParams completionParams) {
 
-        logger.info("Workspace Path: " + workspaceRoot.toString());
         List<CompletionItem> completionItems = new ArrayList<>();
         try {
-            logger.info("URI: " + completionParams.getTextDocument().getUri().substring(7));
-            List<String> lines = Arrays.asList(DocumentTracker.getInstance().getContents(URI.create(completionParams.getTextDocument().getUri())).split("\n"));
+            List<String> lines = Arrays.asList(getDocumentTracker().getContents(URI.create(completionParams.getTextDocument().getUri())).split("\n"));
             String line = lines.get(completionParams.getPosition().getLine());
 
-            logger.info("Working line: " + line);
-            logger.info("Trigger Character: " + completionParams.getContext().getTriggerCharacter());
             String triggerCharacter = completionParams.getContext().getTriggerCharacter();
             if (triggerCharacter.equals("/")) {
-                getPathItems(line, workspaceRoot, completionParams, completionItems);
+                getPathItems(line, completionParams, completionItems);
             } else if (triggerCharacter.equals(":")) {
-                getBuildItems(line, workspaceRoot, completionParams, completionItems);
+                getBuildTargets(line, completionParams, completionItems);
             }
 
-        } catch (IOException e) {
-            logger.error("Hit exception");
-            logger.error(Logging.stackTraceToString(e));
         } catch (Exception e) {
-            // TODO: Find the cause of the completion null pointer exception and error check accordingly
-            logger.error("Hit the catch for generic exceptions");
             logger.error(Logging.stackTraceToString(e));
-        }
+        }// TODO: Find the cause of the completion null pointer exception and error check accordingly
```

Comment:
Did you mean to leave this TODO? I'm not sure if it is something that must be fixed during this sprint.